About Page: Convert to an About section overview that links to several new pages within the about APG section

mcking65 opened the pull request:
Implements #2423.

[Preview of new About section](https://deploy-preview-207--aria-practices.netlify.app/aria/apg/about/)

___
WAI Preview Link [failed to build](https://github.com/w3c/wai-aria-practices/runs/12526577689?check_suite_focus=true) on 'Update site files' step. _(Last tried on Wed, 05 Apr 2023 02:40:13 GMT)._

mcking65 commented:
@isaacdurazo 

The about page in this branch provides draft content for the new about page design; it should work well with your new list view. The subpages of the about section, e.g. introduction, will use the design of the practices pages, e.g., [Providing Accessible Names and Descriptions | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/).

a11ydoer commented:
Note: adding new APG pages like list view and supporting url need to be implemented at wai repo so that preview url can be available.

mcking65 commented:
@alflennik  @howard-e 

Now that I have conflicts resolved and all tests passing, how do I trigger a new attempt to build a preview?

alflennik commented:
Hi @mcking65 the content type referred to here is not HTML, this error is referring to the build system encountering content that doesn't fit a known pattern and therefore cannot be processed. Some content types are "pattern", "example", "practices", "asset" and so on.

Since previously the about page was just a single file, the build script is currently only aware of that one file. With this PR it will become a series of files and the build script will need to be updated to categorize them correctly (and show them).

nschonni commented on content/about/introduction/introduction.html:
```diff
@@ -0,0 +1,112 @@
+        </p>
+        <p>
+          The APG does not specify normative requirements and thus does not have a conformance model.
+          The <a href="https://www.w3.org/groups/tf/aria-practices">ARIA Authoring Practices Task Force</a>, which is part of the <a href=""https://www.w3.org/groups/wg/aria">ARIA Working Group</a>,
```
Two double quotes broke the parsing
```suggestion
          The <a href="https://www.w3.org/groups/tf/aria-practices">ARIA Authoring Practices Task Force</a>, which is part of the <a href="https://www.w3.org/groups/wg/aria">ARIA Working Group</a>,
```

nschonni commented on content/about/introduction/introduction.html:
```diff
@@ -0,0 +1,171 @@
+          It is important to understand, however, that the APG pattern library is not intended to serve as a catalog of all possible, valid, and potentially usable ways of building accessible user interfaces.
+        </p>
+        <p>
+          So, some developers may find it necessary to go beyond  simply combining various APG patterns to creat an experience that supports their UI requirements.
```
```suggestion
          So, some developers may find it necessary to go beyond  simply combining various APG patterns to create an experience that supports their UI requirements.
```

nschonni commented on content/about/about.html:
```diff
@@ -15,285 +15,51 @@
-            the entire Authoring Practices Task Force and also benefits from many people throughout the open source
-            community who both contribute significant work and provide valuable feedback, special thanks goes to the
-            following people who provided distinctly large portions of the content and code in version 1.1.
+            Ffacilitate success when applying APG guidance by developing a solid understanding of what ARIA is, what accessibility semantics are, and  their purpose and limitations with this section about
```
```suggestion
            Facilitate success when applying APG guidance by developing a solid understanding of what ARIA is, what accessibility semantics are, and  their purpose and limitations with this section about
```

nschonni commented on content/about/introduction/introduction.html:
```diff
@@ -0,0 +1,171 @@
+        <p>
+          Some guidance provided by the APG is not covered by any normative specification.
+          Most notably, the patterns include keyboard interaction models that reflect a combination of one normative requirement specified by ARIA and conventions established by popular operating systems and software.
+          That is, in the section on <a href="#managingfocus">Managing focus</a>, ARIA specifies that the <kbd>Tab</kbd> key should not move focus inside of certain types of widgets.
```
The "Managing Focus" link points to an ID `managingfocus` that doesn't exist in the page

alflennik commented on content/about/Acknowledgements/acknowledgements.html:
```diff
@@ -0,0 +1,153 @@
+<!DOCTYPE html>
```
This file is in the "Acknowledgements" category, which is capitalized, but I think the directory shouldn't be capitalized.

mcking65 replied:
fixed

alflennik commented on content/about/introduction/aria-basics.html:
```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
```
My suggestion with this file is that it be given its own folder like the other about pages, that way the content structure is similar across all the about pages as well as across the pattern and practice pages.

mcking65 replied:
When I put aria-basics.html and introduction.html in the same directory, I was thinking of them both as part of introduction. However, I now see that one is introduction to APG while other is more introduction to ARIA. So, per your suggestion, I separated them.

In the future, there could be additional files or content under aria-basics that is related to the fundamentals of ARIA itself.

alflennik reviewed (changes requested):
Hi @mcking65 as I work on integrating this into the website, I have a couple comments about the structure of the content. I like the new additions by the way!

alflennik commented:
The about restructure as it will look on the website can be previewed here: https://deploy-preview-207--aria-practices.netlify.app/aria/apg/about/

andreancardona commented:
All the content looks good on my end - just deleted some unnecessary spaces! 🚀

andreancardona commented on content/about/acknowledgements/acknowledgements.html:
```diff
@@ -0,0 +1,153 @@
+          <li>Philipp Rudloff</li>
+          <li>Joseph Scheuhammer</li>
+          <li>Nick Schonning</li>
+          <li>thomascorthals</li>
```
Is this a user name or full name? `thomascorthals`

alflennik reviewed (approved):
Thank you for addressing my feedback!

mcking65 commented:
@alflennik, my latest commits are not showing up in w3c/wai-aria-practices#207, so the preview is not up-to-date.

alflennik commented:
@mcking65 I've updated the existing preview link with the latest changes.

mcking65 commented:
@alflennik 

Could you please update the preview again? I just addressed the unresolved feedback and fixed a sentence where I made a rather embarrassing editorial blunder.

alflennik commented:
@mcking65 okay, it's updated.

a11ydoer reviewed (approved):
Contents looks good. I liked the conciseness of role, properties and state in "aria basics" page and the table of "How the APG is organized" in "about page".

a11ydoer commented:
@mcking65 @alflennik The review is done. Thanks so much for great work.

a11ydoer commented:
@jnurthen also reviewed this and he said that he did not see any issues.
